whctools/utils.py

```diff
@@ -1,50 +1,59 @@
-from django.utils import timezone
-
-from allianceauth.services.hooks import get_extension_logger
-from app_utils.logging import LoggerAddTag
-
-from whctools import __title__
-from whctools.models import Acl, ACLHistory, Applications
-
-logger = LoggerAddTag(get_extension_logger(__name__), __title__)
-
-def remove_character_from_acl(char_id, acl_name, from_state, to_state, reason):
-    """Helper function to remove a character from an acl"""
-
-
-    acl_object = Acl.objects.filter(pk=acl_name)
-    if acl_object:
-        characters = acl_object[0].characters.all()
-        logger.debug(f"DEBUGDEBUGDEBUG {char_id}")
-        for char in characters:
-            if char.character_id == char_id:
-                logger.debug(f"Removing {char.character_name} form {acl_name} - setting to {Applications.MembershipStates(to_state).name} for {reason.name}")
-                acl_object[0].characters.remove(char)
-                history_entry = ACLHistory(
-                    character=char,
-                    date_of_change=timezone.now(),
-                    old_state = from_state,
-                    new_state = to_state, 
-                    #reason_for_change = reason,
-                    changed_by = "ToDo",
-                    acl=acl_object[0]
-                )
-                history_entry.save()
-                acl_object[0].changes.add(history_entry)
-
-def add_character_to_acl(acl_name, eve_character, old_state, new_state, reason):
-    logger.debug(f"Adding {eve_character.character_name} to {acl_name} - setting to {Applications.MembershipStates(new_state).name} for {reason.name}")
-    acl_obj = Acl.objects.get(pk=acl_name)
-    if acl_obj:
-        acl_obj.characters.add(eve_character)
-        history_entry = ACLHistory(
-                character=eve_character,
-                date_of_change=timezone.now(),
-                old_state=old_state,
-                new_state=new_state,
-                #reason_for_change=reason,
-                changed_by="Acceptance (todo)",
-                acl=acl_obj
-            )
-        history_entry.save()
-        acl_obj.changes.add(history_entry)
\ No newline at end of file
+from django.utils import timezone

```

Comment:
pretty sure this file and the form.py file "completely replaced" because i used a git stash to pop changes over and it just overwrote it 🤷

Reply:
Yeah that can happen